### PR TITLE
Provide better alternative.

### LINKED ISSUE #3143: Bad Event clock examples
In the chapters _Operator 16.11 shiftSample_ and _Operator 16.12 backSample_ the following exemplary code is used to create event clocks:

```
//  _Operator 16.11 shiftSample_ 
Clock u = Clock(sin(2 * pi * time) > 0, startInterval = xx)

//  _Operator 16.12 backSample_ 
Clock u = Clock(sin(2 * pi * time) > 0, startInterval = 0.0);
```

I suggest to replace these examples, as they are potentially dangerous and give users a wrong impression, how event clocks shall be used. Of course you can do that, but such event clocks can easily break and give unexpected results (at least in Dymola and OpenModelica).

Just simulate this example with default settings (Dassl, equidistant time grid on, 500 output points) in Dymola 2022x or OpenModelica v1.18.1:

```
model EventClock
  Clock clock = Clock(sin(2*Modelica.Constants.pi*10000*time)>0);
  Integer counter(start=0);
equation 
  when clock then
    counter = previous(counter) + 1;
  end when;
end EventClock;
````

`counter` should be 10000 at `t=1s`, but Dymola 2022x gives 3910 and OpenModelica v1.18.1 gives 0.

The reason for that is that the tools fail to detect the zero crossing due to large integrator steps _(when the number of output points are increased we get correct results in both tools, as we force the integrator to take smaller steps)_.

### PR DESCRIPTION
**Pull request by HansOlsson**
Closes #3143
Having examples that we know lead to numerical issues isn't a good idea, especially as there's an equivalent formulation without the problems - and we know that many just copy the examples from the specification as a starting point.

### CODE REVIEW

File: chapters/synchronous.tex [hunk 1]
```diff
@@ -720,11 +720,13 @@ \subsection{Sub-Clock Conversion Operators}\label{sub-clock-conversion-operators
 Clock u  = Clock(3, 10);            // ticks: 0, 3/10, 6/10, $\ldots$
 Clock y1 = shiftSample(u, 1, 3);    // ticks: 1/10, 4/10, $\ldots$
 // Event clock
-Clock u = Clock(sin(2 * pi * time) > 0, startInterval = 0.0);
+Integer revolutions=integer(time);
```
Comment (henrikt-ma) [inline chapters/synchronous.tex:723]: Space:
```suggestion
Integer revolutions = integer(time);
```

File: chapters/synchronous.tex [hunk 2]
```diff
@@ -751,7 +753,8 @@ \subsection{Sub-Clock Conversion Operators}\label{sub-clock-conversion-operators
 Clock y4 = shiftSample(u, 2, 3);  // ticks: 2/10, 5/10, $\ldots$
 Clock y5 = backSample(y4, 1, 3);  // ticks: 1/10, 4/10, $\ldots$
 // Event clock
-Clock u = Clock(sin(2 * pi * time) > 0, startInterval = xx)
+Integer revolutions=integer(time);
```
Comment (henrikt-ma) [inline chapters/synchronous.tex:756]: Space:
```suggestion
Integer revolutions = integer(time);
```

File: chapters/synchronous.tex [hunk 3]
```diff
@@ -720,11 +720,13 @@ \subsection{Sub-Clock Conversion Operators}\label{sub-clock-conversion-operators
...
 Clock y1 = shiftSample(u, 1, 3);    // ticks: 1/10, 4/10, $\ldots$
 // Event clock
-Clock u = Clock(sin(2 * pi * time) > 0, startInterval = 0.0);
+Integer revolutions=integer(time);
+Clock u = Clock(change(revolutions), startInterval = 0.0);
                                     // ticks: 0.0, 1.0, 2.0, 3.0, $\ldots$
 Clock y1 = shiftSample(u, 2);       // ticks: 2.0, 3.0, $\ldots$
 Clock y2 = shiftSample(u, 2, 3);    // error (resolution must be 1)
 \end{lstlisting}
+Note: The events generated by \lstinline!change(revolutions)! are at the same time as \lstinline!sin(2 * pi * time) > 0!, but it is numerically more robust.
```
Comment (henrikt-ma) [inline chapters/synchronous.tex:729]: I don't see the need to include this note now that both occurrences of `sin(2 * pi * time) > 0` have been removed.
```suggestion
```

**Review by henrikt-ma — CHANGES_REQUESTED** [review]
While I didn't see a strict need to change anything in view of the low frequency of the zero crossing sine, I don't see a problem with using `change(revolutions)` either.

My only request for change is that we then also remove the last remaining trace of the sine zero crossing.

File: chapters/synchronous.tex (hunk 3 above)
Comment (eshmoylova) [reply chapters/synchronous.tex:729]: I agree. I think it is confusing to mention `sin(2 * pi * time) > 0` when it does not occur anywhere now.

**Review by eshmoylova — CHANGES_REQUESTED** [review]

**Review by eshmoylova — APPROVED** [review]

**Review by henrikt-ma — COMMENTED** [review]
I don't really have anything to object to, but I note that all of this came out of an interesting discussion about high frequency zero crossing functions, and the proposed change is only sweeping that under the rug.  On the other hand, as the topic is a difficult numerical issue, I don't want to insist on trying to find the right formulations to put in the specification either.

**Comment by HansOlsson** [comment]
> My only request for change is that we then also remove the last remaining trace of the sine zero crossing.

That is now done.

**Review by henrikt-ma — COMMENTED** [review]
> > My only request for change is that we then also remove the last remaining trace of the sine zero crossing.
>
> That is now done.

Yes, I noted, and that's why I only have a comment regarding the potential loss of the interesting discussion about problems with high frequency signals in zero crossings. As long as there's some sort of agreement that we don't want to add anything regarding that, I'm fine with the current state.

**Review by henrikt-ma — APPROVED** [review]
Approving, as there didn't seem to be that much interest in adding some general remark about the difficulty to handle high frequency zero crossing functions.

**Pull request merged**